packages/databyss-ui/components/Editor/slate/ContentEditable.js

```diff
@@ -189,8 +208,41 @@ const SlateContentEditable = ({
 
     if (_nextText !== _prevText) {
       const block = _nextEditableState.value.anchorBlock
-      const jsonBlockValue = { ...block.toJSON(), key: block.key }
-      const ranges = getRangesFromBlock(jsonBlockValue).ranges
+      const ranges = getBlockRanges(block)
+
+      // if not atomic get range and check for location
+      if (
+        !isAtomicInlineType(
+          _nextEditableState.value.document.getNode(activeBlockId).type
+        )
+      ) {
+        const locationLength = ranges.reduce((acc, range) => {
+          if (range.marks.findIndex(m => m === 'location') > -1) {
+            return range.length + acc
+          }
+          return acc
+        }, 0)
+
+        // if type LOCATION is set check for non LOCATION type
```

Comment:
This condition doesn't make sense to me. Can you write a test that demonstrates the use case?